CNV#47262 (#92182)
openshift/openshift-docs: modules/virt-optimizing-clone-performance-at-scale-in-openshift-data-foundation.adoc

// Module included in the following assemblies:
//
// * virt/virtual_machines/creating_vms_custom/virt-creating-vms-by-cloning-pvcs.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-optimizing-clone-performance-at-scale-in-openshift-data-foundation_{context}"]

> = Optimizing Clone Performance at Scale in {rh-storage}

Reviewer: https://redhat-documentation.github.io/supplementary-style-guide/#titles-and-headings

> = Optimizing Clone Performance at Scale in OpenShift Data Foundation

Reviewer 2: ```suggestion
= Optimizing Clone Performance at Scale in {rh-storage}
```

> When you use {rh-storage}, the storage profile configures the default cloning strategy as `csi-clone`. However, this method has limitations as shown in the link below. After a certain number of clones are created from a persistent volume claim (PVC), a background flattening process begins. This process can significantly reduce clone creation performance at scale.

Reviewer: ```suggestion
When you use {rh-storage}, the storage profile configures the default cloning strategy as `csi-clone`. However, this method has limitations, as shown in the following link. After a certain number of clones are created from a persistent volume claim (PVC), a background flattening process begins, which can significantly reduce clone creation performance at scale.
```

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#location-of-interface-elements

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism

To improve performance when creating hundreds of clones from a single source PVC, use the `VolumeSnapshot`cloning method instead of the default `csi-clone` strategy.

> 

Reviewer: ```suggestion
.Procedure
```

Reviewer: https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

> To use this method, first create a `VolumeSnapshot` of the source image.

Reviewer: ```suggestion
Create a `VolumeSnapshot` custom resource (CR) of the source image by using the following content:
```

> When you use OpenShift Data Foundation, the storage profile configures the default link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/virtualization/storage#virt-customizing-storage-profile-default-cloning-strategy_virt-configuring-storage-profile[cloning strategy] as `csi-clone`. However, this method has link:https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.15/html/managing_and_allocating_storage_resources/volume-cloning_rhodf#creating-a-clone_rhodf[limitations]. After a certain number of clones are created from a persistent volume claim (PVC), a background flattening process begins. This process can significantly reduce clone creation performance at scale.

Reviewer 3: @Vaishali-gif-rh please check the version to ensure this is correct. My initial thought is that we should reference the latest version. I reviewed the latest version of 4.18 and the content is the same and accurate.

Author: @ctomasko - Noted. Thanks for mentioning this!

[source,yaml]
----
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshot
metadata: 
  name: golden-volumesnapshot
  namespace: golden-ns
spec:
  volumeSnapshotClassName: ocs-storagecluster-rbdplugin-snapclass
  source:
    persistentVolumeClaimName: golden-snap-source

> 

Reviewer: ```suggestion
```

----

> Then, reference the `VolumeSnapshot` as the source for the `DataVolume clone`.

Reviewer: ```suggestion
. Add the  `spec.source.snapshot` stanza to reference the `VolumeSnapshot` as the source for the `DataVolume clone`:
```

> 

Reviewer: ```suggestion
```

[source,yaml]
----

> spec:

Reviewer: ```suggestion
# ...
spec:
```

To improve performance when creating hundreds of clones from a single source PVC, use the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/storage/using-container-storage-interface-csi#persistent-storage-csi-snapshots-overview_persistent-storage-csi-snapshots[`VolumeSnapshot`] cloning method instead of the default `csi-clone` strategy.

To use this method, first create a `VolumeSnapshot` of the source image.
[source,yaml]
----
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshot
metadata: 
  name: golden-volumesnapshot
  namespace: golden-ns
spec:
  volumeSnapshotClassName: ocs-storagecluster-rbdplugin-snapclass
  source:
    persistentVolumeClaimName: golden-snap-source

----

Then, reference the `VolumeSnapshot` as the source for the `DataVolume clone`.

[source,yaml]
----
spec:
    source:
    snapshot:
        namespace: golden-ns
        name: golden-volumesnapshot

> ----

Reviewer 4: spacing is slightly off here^^, should be:
```
    spec:
     source:
       snapshot:
        namespace: golden-ns
        name: golden-volumesnapshot
```
